src/graph/graph.c

```diff
@@ -260,16 +260,78 @@ void Graph_SetMatrixPolicy
 // synchronize and resize all matrices in graph
 void Graph_ApplyAllPending
 (
-	Graph *g
+	Graph *g,
+	bool force_flush
+) {
+	ASSERT(g != NULL);
+
+	uint       n  =  0;
+	RG_Matrix  M  =  NULL;
+
+	M = Graph_GetAdjacencyMatrix(g, false);
+	RG_Matrix_wait(M, force_flush);
+
+	n = array_len(g->labels);
+	for(int i = 0; i < n; i ++) {
+		M = Graph_GetLabelMatrix(g, i);
+		RG_Matrix_wait(M, force_flush);
+	}
+
+	n = array_len(g->relations);
+	for(int i = 0; i < n; i ++) {
+		M = Graph_GetRelationMatrix(g, i, false);
+		RG_Matrix_wait(M, force_flush);
+	}
+}
+
+bool Graph_Pending
```

Comment:
why not return immediately when any matrix is pending?

Reply:
👍